.github/workflows/build.yml

```diff
@@ -23,3 +23,6 @@ jobs:
       - uses: gradle/gradle-build-action@v2
       - name: Build
         run: ./gradlew :server:build :shared:build -PjavaVersion=${{ matrix.java }}
+      - uses: gradle/gradle-build-action@v2
```

Comment:
You're right @fwcd, it's duplicated because the `build` task depends on `check`, which depends on `detekt`. I'm wondering how we should go about this now. It's great to be more explicit about the steps that we're performing in the build as a part of our CI checks, but I also see a lot of value in keeping the `check` task dependent on the static analysis detekt performs. Do we know what any other projects do (or if there's a best practice around this)? If not I'll take a look at some popular libraries on GitHub, like ktlint, and see what they do.

Reply:
What I meant is that

```
- uses: gradle/gradle-build-action@v2
```

is a fully distinct step from

```
- name: Build
   run: ./gradlew :server:build :shared:build -PjavaVersion=${{ matrix.java }}
```

and we probably don't need to run the former one twice. In fact, we might not even need it at all, since the Gradle wrapper should already take care of installing Gradle.

(yes, YAML syntax might be a bit subtle here)

Reply:
Ah I see what you mean now. 

Should we do:

```
- name: Build
   run: ./gradlew build -PjavaVersion=${{ matrix.java }}
```

To capture the root level configurations as well? 
Is there a reason we don't build the `grammar` subproject on the pipeline? I haven't seen the discussion about it just yet

Reply:
The `grammar` subproject doesn't really need to be built, since it's just a bunch of JSON files. The reason we have a Gradle subproject for it is to have a convenient `distZip` task for packaging up releases.